src/torchmetrics/functional/classification/group_fairness.py

```diff
@@ -0,0 +1,374 @@
+    be ``0, 1, ..., (num_groups - 1)``.
+    - The group tensor should be dtype long.
+    """
+    if torch.max(groups) > num_groups:
+        raise ValueError(
+            f"The largest number in the groups tensor is {torch.max(groups)}, which is larger than the specified",
+            f"number of groups {num_groups}. The group identifiers should be ``0, 1, ..., (num_groups - 1)``.",
+        )
+    if groups.dtype != torch.long:
+        raise ValueError(f"Excpected dtype of argument groups to be long, not {groups.dtype}.")
+
+
+def _groups_format(groups: torch.Tensor) -> torch.Tensor:
+    """Reshape groups to correspond to preds and target."""
+    groups = groups.reshape(groups.shape[0], -1)
+
+    return groups
+
+
+def _binary_groups_stat_scores(
+    preds: torch.Tensor,
+    target: torch.Tensor,
+    groups: torch.Tensor,
+    num_groups: int,
+    threshold: float = 0.5,
+    ignore_index: Optional[int] = None,
+    validate_args: bool = True,
+) -> List[Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]]:
+    """Computes the number of true positives, false positives, true negatives, and false negatives for binary
+    classification by group.
+
+    Related to `Type I and Type II errors`_.
+    """
+    if validate_args:
+        _binary_stat_scores_arg_validation(threshold, "global", ignore_index)
+        _binary_stat_scores_tensor_validation(preds, target, "global", ignore_index)
+        _groups_validation(groups, num_groups)
+
+    preds, target = _binary_stat_scores_format(preds, target, threshold, ignore_index)
+    groups = _groups_format(groups)
+
+    indexes, indices = torch.sort(groups.squeeze(1))
+    preds = preds[indices]
+    target = target[indices]
+
+    split_sizes = _flexible_bincount(indexes).detach().cpu().tolist()
+
+    group_preds = list(torch.split(preds, split_sizes, dim=0))
+    group_target = list(torch.split(target, split_sizes, dim=0))
+
+    group_stats = [_binary_stat_scores_update(group_p, group_t) for group_p, group_t in zip(group_preds, group_target)]
+
+    return group_stats
+
+
+def _groups_reduce(
+    group_stats: List[Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]]
+) -> Dict[str, torch.Tensor]:
+    """Compute rates for all the group statistics."""
+    return {f"group_{group}": torch.stack(stats) / torch.stack(stats).sum() for group, stats in enumerate(group_stats)}
+
+
+def _groups_stat_transform(
+    group_stats: List[Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]]
+) -> Dict[str, torch.Tensor]:
+    """Transform group statistics by creating a tensor for each statistic."""
+    return {
+        "tp": torch.cat([stat[0].unsqueeze(0) for stat in group_stats]),
+        "fp": torch.cat([stat[1].unsqueeze(0) for stat in group_stats]),
+        "tn": torch.cat([stat[2].unsqueeze(0) for stat in group_stats]),
+        "fn": torch.cat([stat[3].unsqueeze(0) for stat in group_stats]),
```

Comment:
pretty sure that stack = unsqueeze + cat
```suggestion
        "tp": torch.stack([stat[0] for stat in group_stats]),
        "fp": torch.stack([stat[1] for stat in group_stats]),
        "tn": torch.stack([stat[2] for stat in group_stats]),
        "fn": torch.stack([stat[3] for stat in group_stats]),
```